qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,304 @@
+    OptimizationResultStatus,
+    SolutionSample,
+)
+from qiskit_optimization.converters import QuadraticProgramConverter, QuadraticProgramToQubo
+from qiskit_optimization.problems import QuadraticProgram, Variable
+
+from .quantum_random_access_encoding import QuantumRandomAccessEncoding
+from .rounding_common import RoundingContext, RoundingResult, RoundingScheme
+from .semideterministic_rounding import SemideterministicRounding
+
+
+class QuantumRandomAccessOptimizationResult(OptimizationResult):
+    """Result of Quantum Random Access Optimization procedure."""
+
+    def __init__(
+        self,
+        *,
+        x: list[float] | np.ndarray,
+        fval: float,
+        variables: list[Variable],
+        status: OptimizationResultStatus,
+        samples: list[SolutionSample],
+        encoding: QuantumRandomAccessEncoding,
+        relaxed_fval: float,
+        relaxed_result: MinimumEigensolverResult,
+        rounding_result: RoundingResult,
+    ) -> None:
+        """
+        Args:
+            x: The optimal value found by ``MinimumEigensolver``.
+            fval: The optimal function value.
+            variables: The list of variables of the optimization problem.
+            status: The termination status of the optimization algorithm.
+            samples: The list of ``SolutionSample`` obtained from the optimization algorithm.
+            encoding: The encoding used for the optimization.
+            relaxed_fval: The optimal function value of the relaxed problem.
+            relaxed_result: The result obtained from the underlying minimum eigensolver.
+            rounding_result: The rounding result.
+        """
+        super().__init__(
+            x=x,
+            fval=fval,
+            variables=variables,
+            status=status,
+            raw_results=None,
+            samples=samples,
+        )
+        self._encoding = encoding
+        self._relaxed_fval = relaxed_fval
+        self._relaxed_result = relaxed_result
+        self._rounding_result = rounding_result
+
+    @property
+    def encoding(self) -> QuantumRandomAccessEncoding:
+        """The encoding used for the optimization."""
+        return self._encoding
+
+    @property
+    def relaxed_fval(self) -> float:
+        """The optimal function value of the relaxed problem."""
+        return self._relaxed_fval
+
+    @property
+    def relaxed_result(
+        self,
+    ) -> MinimumEigensolverResult:
+        """The result obtained from the underlying minimum eigensolver."""
+        return self._relaxed_result
+
+    @property
+    def rounding_result(self) -> RoundingResult:
+        """The rounding result."""
+        return self._rounding_result
+
+
+class QuantumRandomAccessOptimizer(OptimizationAlgorithm):
+    """Quantum Random Access Optimizer class."""
+
+    def __init__(
+        self,
+        min_eigen_solver: MinimumEigensolver,
+        max_vars_per_qubit: int = 3,
+        rounding_scheme: RoundingScheme | None = None,
```

Comment:
```     
        self,
        min_eigen_solver: MinimumEigensolver,
        *,
        max_vars_per_qubit: int = 3,
        rounding_scheme: RoundingScheme | None = None,
```
I cannot recall what item before I mentioned about perhaps thinking about adding keyword enforcement. My suggestion would be like this above. Also I seem to remember commenting about penalty - may again have been a different place.  Is there a reason why we do not have it as an explicit argument - its rather obscure for an end user if its set via kwargs especially since its not even mentioned.

Update: Ok I went back and found what I said before, and it was here

>Should we make penalty and beyond keyword only arguments - this way its easier to extend/change in the future as we don;t have to cater to positional usage of everything.

By `make penalty and beyond keyword only argument` I simply meant adding `*,` like I did in the suggestion above to ensure they need to be supplied as keyword args, not adding a kwargs. Sorry if it was no so clear before. 
I guess I said penalty and beyond before - that would change where we decide to put the `*,`  - with penalty back in the list explicitly (as in which ones do we allow positionally). Thus this below is was more what I was imagining before

```
       self,
        min_eigen_solver: MinimumEigensolver,
        max_vars_per_qubit: int = 3,
        rounding_scheme: RoundingScheme | None = None,
        *,
        penalty: XXXX | None = None   (not sure penalty type but you get the picture I hope)
```

Update: I see in looking further the SemiDeterministicRounding is an example like this - it made all the parameters keyword only